src/flow.ts

```diff
@@ -831,17 +839,22 @@ export class Flow {
       }
     }
 
-    // Only the most right flow will have an effect
-    // on the resulting flow.
-    const rightFieldFlags = right.fieldFlags;
-    const rightKeys = Map_keys(rightFieldFlags);
 
-    for (let _values = Map_values(rightFieldFlags), i = 0, k = _values.length; i < k; ++i) {
-      const rightValue = unchecked(_values[i]);
-      const currentKey = unchecked(rightKeys[i]);
+    if (left.fieldFlags && right.fieldFlags && right.fieldFlags.size > 0) {
```

Comment:
This will currently fail to compile to Wasm because it deals with `object && object && bool`. More of a bug that will be fixed eventually, but explicitly comparing vs `!== null` will work today.